tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -2448,10 +2446,14 @@ std::optional<Value> convertStridedSliceOp(
   }
 
   // Step 2: reshape the sliced array
-  SmallVector<int64_t> a2_shape(input_rank * 2);
+  SmallVector<int64_t> a2_shape;
   for (int i = 0; i < input_rank; ++i) {
-    a2_shape[i * 2 + 0] = a1_size[i] == -1 ? -1 : a1_size[i] / abs(strides[i]);
-    a2_shape[i * 2 + 1] = abs(strides[i]);
+    int64_t abs_stride_i = abs(strides[i]);
+    a2_shape.push_back(a1_size[i] == -1 ? -1 : a1_size[i] / abs_stride_i);
+    if (abs_stride_i != 1) {
+      // only add a stride dimension if strides[i] != 1
```

Comment:
This is 1 and -1 right, why are both allowed here?

Reply:
good catch. let me fix this.

Reply:
turned out the original code always added a stride dimension of abs(stride[i]).
the refactor in this patch is simply to filter out cases where abs(stride[i]) is 1 because there is no need to add a extra dimension of value 1.